classes/task/ReviewReminder.inc.php

```diff
@@ -56,70 +62,41 @@ public function sendReminder($reviewAssignment, $submission, $context, $reminder
             $accessKeyManager = new AccessKeyManager();
 
             // Key lifetime is the typical review period plus four weeks
             $keyLifetime = ($context->getData('numWeeksPerReview') + 4) * 7;
             $accessKey = $accessKeyManager->createKey($context->getId(), $reviewer->getId(), $reviewId, $keyLifetime);
-            $reviewUrlArgs = array_merge($reviewUrlArgs, ['reviewId' => $reviewId, 'key' => $accessKey]);
-        }
 
-        $application = Application::get();
-        $request = $application->getRequest();
-        $dispatcher = $application->getDispatcher();
-        $submissionReviewUrl = $dispatcher->url($request, PKPApplication::ROUTE_PAGE, $context->getPath(), 'reviewer', 'submission', null, $reviewUrlArgs);
-
-        // Format the review due date
-        $reviewDueDate = strtotime($reviewAssignment->getDateDue());
-        $dateFormatShort = $context->getLocalizedDateFormatShort();
-        if ($reviewDueDate === -1 || $reviewDueDate === false) {
-            // Default to something human-readable if no date specified
-            $reviewDueDate = '_____';
-        } else {
-            $reviewDueDate = strftime($dateFormatShort, $reviewDueDate);
-        }
-        // Format the review response due date
-        $responseDueDate = strtotime($reviewAssignment->getDateResponseDue());
-        if ($responseDueDate === -1 || $responseDueDate === false) {
-            // Default to something human-readable if no date specified
-            $responseDueDate = '_____';
-        } else {
-            $responseDueDate = strftime($dateFormatShort, $responseDueDate);
+            $reviewUrlArgs = ['submissionId' => $reviewAssignment->getSubmissionId(), 'reviewId' => $reviewId, 'key' => $accessKey];
+            $submissionReviewUrl = $dispatcher->url($request, PKPApplication::ROUTE_PAGE, $context->getPath(), 'reviewer', 'submission', null, $reviewUrlArgs);
+            $mailable->addData(['submissionReviewUrl' => $submissionReviewUrl]);
         }
-        $paramArray = [
-            'recipientName' => $reviewer->getFullName(),
-            'recipientUsername' => $reviewer->getUsername(),
-            'reviewDueDate' => $reviewDueDate,
-            'responseDueDate' => $responseDueDate,
-            'signature' => $context->getData('contactName') . "\n" . $context->getLocalizedName(),
+
+        $oldVariables = [
             'passwordResetUrl' => $dispatcher->url($request, PKPApplication::ROUTE_PAGE, $context->getPath(), 'login', 'resetPassword', $reviewer->getUsername(), ['confirm' => Validation::generatePasswordResetHash($reviewer->getId())]),
-            'reviewAssignmentUrl' => $submissionReviewUrl,
             'messageToReviewer' => __('reviewer.step1.requestBoilerplate'),
             'abstractTermIfEnabled' => ($submission->getLocalizedAbstract() == '' ? '' : __('common.abstract')),
         ];
-        $email->assignParams($paramArray);
+        $mailable->addData($oldVariables);
```

Comment:
Do these variables need to be added to the description of supported variables for each mailable?

Reply:
No need to add them. Latter two variables aren't used for a long time and the former has an analogue variable `passwordLostUrl` with the description provided through `ContextEmailVariable`.

Reply:
I see that `passwordResetUrl` is used in the password reset confirmation email, adding a description anyway.